app/game.js

```diff
@@ -0,0 +1,86 @@
+function Game(results) {
+  this._score = 0;
+  this._pins = 10;
+  this.bonus = false;
+  this._balls = 20;
+  this.results = results;
+};
+
+Game.prototype.start = function () {
+  for(i=0; i < 9*2; i+=2){
+    this.frame(i);
+  };
+  this.tenthFrame();
+};
+
+Game.prototype.frame = function (i) {
+  var pins_down = this.getRoll(i);
+  if (this.bonus && pins_down != 0) {
+    this.applyBonus(pins_down);
+  };
+  if (pins_down < this._pins) {
+    pins_down += this.getRoll(i+1);
+    if (this.bonus) {
+      this.applyBonus(pins_down);
+    };
+  };
+  this._score += pins_down;
+  if (pins_down === this._pins) this.setBonus();
+};
+
+Game.prototype.getRoll = function(i) {
```

Comment:
Watch out for this function's name. It implies that it just returns a value, but it also changes data.